Source/cielim/Private/SpawnScene.cpp

```diff
@@ -0,0 +1,135 @@
+	//Init spacecraft and light positions
+	FVector3d SpacecraftPosition {0,0,height};
+	FVector3d PointLightPosition {radius,0,height};
+	
+	//Init spacecraft and light rotations
+	FRotator SpacecraftRotation = FRotator::ZeroRotator;
+	FRotator PointLightRotation = FRotator::ZeroRotator;
+
+	//Spawn spacecraft actor
+	Spacecraft = GetWorld()->SpawnActor<ASpacecraft>(BpSpacecraft, SpacecraftPosition, SpacecraftRotation);
+	UE_LOG(LogCielim, Warning, TEXT("Spacecraft Spawned"));
+
+	//Spawn light actor
+	PointLight = GetWorld()->SpawnActor<APointLightActor>(BpPointLight, PointLightPosition, PointLightRotation);
+	UE_LOG(LogCielim, Warning, TEXT("PointLight Spawned"));
+	UE_LOG(LogCielim, Warning, TEXT("Scene Spawned"));
+	
+	Super::BeginPlay();
+	
+}
+
+// Called every frame
+void ASpawnScene::Tick(float DeltaTime)
+{
+	//If the light source has changed directions less than twice... in the horizontal direction
+	if(alpha_flag_changes < 2)
+	{
+		//if the light source is outside the right bound
+		if(alpha > PI / 4)
+		{
+			clockwise_flag = false;
+			alpha_flag_changes++;
+		}
+		//if the light source is outside the left bound
+		else if(alpha < -1 * PI / 4)
+		{
+			clockwise_flag = true;
+			alpha_flag_changes++;
+		}
+		//if the light source is going clockwise
+		if(clockwise_flag)
+		{
+			alpha += 2 * PI / 4 / 200;
+			UpdateHorizontalPointLight(alpha, radius, height);
+		}
+		//if the light source is going counter-clockwise
+		else
+		{
+			alpha -= 2 * PI / 4 / 200;
+			UpdateHorizontalPointLight(alpha, radius, height);
+		}
+	}
+	//If the light source has changed directions less than twice... in the vertical direction
+	else if(beta_flag_changes < 2)
+	{
+		ResetPointLight(radius, height);
+		//if the light source is outside the upper(?) bound
+		if(beta > PI / 4)
+		{
+			clockwise_flag = false;
+			beta_flag_changes++;
+		}
+		//if the light source is outside the lower(?) bound
+		else if(beta < -1 * PI / 4)
+		{
+			clockwise_flag = true;
+			beta_flag_changes++;
+		}
+		//if the light source is going clockwise
+		if(clockwise_flag)
+		{
+			beta += 2 * PI / 4 / 200;
+			UpdateVerticalPointLight(beta, radius, height);
+		}
+		//if the light source is going counter-clockwise
+		else
+		{
+			beta -= 2 * PI / 4 / 200;
+			UpdateVerticalPointLight(beta, radius, height);
+		}
+	}
+
+	//if the light source has gone both ways in the vertical and horizontal directions, reset all the parameters and start over
+	else
+	{
+		alpha_flag_changes = 0;
+		beta_flag_changes = 0;
+		alpha = -1 * PI / 4;
+		beta = -1 * PI / 4;
+		ResetPointLight(radius, height);
+	}
+	
+	Super::Tick(DeltaTime);
+}
+
+void ASpawnScene::UpdateHorizontalPointLight(float a, float r, float h)
+{
+	this->PointLight->UpdateHorizontalPath(a, r, h);
+}
+
+void ASpawnScene::UpdateVerticalPointLight(float b, float r, float h)
+{
+	this->PointLight->UpdateVerticalPath(b, r, h);
+}
+
+void ASpawnScene::ResetPointLight(float r, float h)
+{
+	this->PointLight->ResetPosition(r, h);
+}
+
```

Comment:
A number of extraneous new lines at the end of this file. We'll add pre-commit to this project to help alert us to these things before making the PR. 👍